service/controller/v11/resource/migration/resource.go

```diff
@@ -0,0 +1,83 @@
+// Package migration provides an operatorkit resource that migrates awsconfig CRs
+// to reference the default credential secret if they do not already.
+// It can be safely removed once all awsconfig CRs reference a credential secret.
```

Comment:
I do not understand the migration thingy. Why do we need it? I would expect customers using the BYOC magic in production right now. I am not aware of this. So what do we need to migrate at all? Also, what does `to reference the default credential secret` actually mean?

Reply:
We need it so this change https://github.com/giantswarm/aws-operator/pull/916 can actually work. If you check our installations we have `giantswarm/credential-default` secret everywhere. It's prepared to allow access to current guest AWS accounts. In the future (as soon as v10 and predecessors are removed) we'll remove guest AWS credential flags.